### DISCUSSION TITLE
[Question] Does Powersync store all my database data and history change on Mongodb or Powersync internal storage?

### DISCUSSION THREAD

**Discussion by ducpt-bili** [powersync-ja/powersync-js#294] [Q&A] [answered]
hi Powersync team,
i just want to ask that does Powersync store all my database data and history change on Mongodb 
or Powersync other internal storage? And it is charge on free plan 500Mb data host service right?
Thank you

**Comment by ducpt-bili** [comment]
hi @rkistner , maybe you can answer that. Sorry for a lots of questions, but i'm heavily investigate on Powersync to decide what techstack should our team will use in the next project.

**Comment by rkistner** [comment, marked as answer]
We store the history of changes in MongoDB. That is re-computed from the source database whenever you deploy sync rule changes.

You can see the code for that here: https://github.com/powersync-ja/powersync-service, and a diagram with the architecture for self-hosting is here: https://docs.powersync.com/self-hosting/installation

Our [discord](https://discord.gg/powersync) is also great for questions like this.

**Comment by cahofmeyr** [comment]
FYI, Postgres can now also be used as alternative to MongoDB for storing the "bucket data" (history of changes) https://releases.powersync.com/announcements/introducing-postgres-for-sync-bucket-storage
